src/framework/utils.hpp

```diff
@@ -697,6 +707,105 @@ matrix<T> tensor_product(const matrix<T> &A, const matrix<T> &B) {
   return temp;
 }
 
+template <class T>
+matrix<T> concatenate (const matrix<T> &A, const matrix<T> &B, uint_t axis) {
+  // Works out a concatenate of two matrices A, B
+  // To add B as new rows axis == 0 (vertical)
+  // To add A as new cols axis == 1 (horizontal)
+  if (axis != 0 && axis!= 1) {
+    std::cerr << "MU::concatenate: axis must be 0 or 1" << std::endl;
```

Comment:
We use exceptions for errors that cannot be recovered from. In this case, we just need:
```c++
throw std::invalid_argument("axis must be 0 or 1");
```

Reply:
Done. I changed the error message to "Utils::concatenate: axis must be 0 or 1". I noticed other functions use "MU" rather than "Utils". I am not sure why. Let me know if I should change this message. Same for the next few comments.

Reply:
As this is an exception, we will know where it comes from because the stack is usually embedded in the exception information, so I'd explicitily write: "Utils::concatenate or MU", with the message: "axis must be 0 or 1" is enough.

Reply:
When I intentionally insert an error, all I get is "qasm_simulator: simulation failed" without the stack, and did not see the error message.  When I ran in a different way, I got the python stack and the error message, but not the C++. How do you run so as to see the error stack? is there some option to show this?